docs/io/fits/index.rst

```diff
@@ -309,6 +309,76 @@ would with a dict::
 
     See also :ref:`sphx_glr_generated_examples_io_modify-fits-header.py`.
 
+.. _structural_keywords:
+
+Structural Keywords
+"""""""""""""""""""
+
+An unfortunate design quirk of FITS is that FITS keywords mix up both metadata
+and critical information file structure that is needed to parse the file.
+These *structural* keywords can be silently modified by :mod:`astropy.io.fits` and,
+in general, should not be touched by the user.
+
+The specific set of structural keywords used by the FITS standard varies with HDU type.
+The following table lists which keywords are associated with each HDU type:
+
+.. csv-table:: Structural Keywords
+   :header: "HDU Type", "Structural Keywords"
+   :widths: 20, 20
+
+   "All", "``SIMPLE``, ``BITPIX``, ``NAXIS``"
+   ":class:`PrimaryHDU`", "``EXTEND``"
+   ":class:`ImageHDU`, :class:`TableHDU`, :class:`BinTableHDU`",  "``PCOUNT``, ``GCOUNT``"
+   ":class:`GroupsHDU`", "``NAXIS1``, ``GCOUNT``, ``PCOUNT``, ``GROUPS``"
+   ":class:`TableHDU`, :class:`BinTableHDU`", "``TFIELDS``, ``TFORM``, ``TBCOL``"
+
+Structural keywords are checked and updated in four different places:
+
+1. When reading a file
+2. When setting the associated data
+3. When setting the :class:`Header` object
+4. When writing a file
+
+This can often cause confusion because any handwritten values users might assign
+to those keywords will be overwrittten. Consider the following examples:
```

Comment:
Again I would remove the first part of the sentence and use "Any manually set values to these keywords ..."